frame/nfts/src/functions.rs

```diff
@@ -138,6 +145,9 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			CollectionAccount::<T, I>::remove(&collection_details.owner, &collection);
 			T::Currency::unreserve(&collection_details.owner, collection_details.total_deposit);
 			CollectionMaxSupply::<T, I>::remove(&collection);
+			CollectionConfigOf::<T, I>::remove(&collection);
+			#[allow(deprecated)]
+			ItemConfigOf::<T, I>::remove_prefix(&collection, None);
```

Comment:
The PoV Size of this is potentially unlimited. This is basically the same problem as https://github.com/paritytech/substrate/pull/12310 which favours a multi-step approach.

Reply:
Maybe one of the standard Error enums should be `Error::IncompletePleaseCallAgain` ?

Reply:
The witness should probably contain an upper limit to bound the weight.